cmd/k8s-operator/proxygroup.go

```diff
@@ -601,6 +634,27 @@ func (r *ProxyGroupReconciler) getDeviceInfo(ctx context.Context, pg *tsapi.Prox
 type nodeMetadata struct {
 	ordinal     int
 	stateSecret *corev1.Secret
+	pod         *corev1.Pod // nil if there is no Pod (yet)
```

Comment:
This is a massive and mutable struct to hold a reference to, but all we ultimately use in the consuming function (`proxyCapVer`) is the podUID - could we maybe update it to store and thread through `podUID` instead of `pod`? More just for code readability than any micro-optimisation

Reply:
yeah that's better, memory consumption too since these would be copies from the cached Pods
we should also do metadata only caching for these maybe
I haven't looked at memory consumption for ages